zkevm-circuits/src/evm_circuit/util/math_gadget.rs

```diff
@@ -200,6 +201,268 @@ impl<F: FieldExt, const N: usize> AddWordsGadget<F, N> {
     }
 }
 
+/// Construction of 2 256-bit words mutiplication and result (modulo 2**256),
+/// which is useful for opcode MUL, DIV, SDIV and xxxMOD
+#[derive(Clone, Debug)]
+pub(crate) struct MulWordsGadget<F> {
+    multiplier: util::Word<F>,
+    multiplicand: util::Word<F>,
+    product: util::Word<F>,
+    //here we execute a multi-limbs multiplication, see spec or
```

Comment:
It'd be better you add the document directly to each variable instead of forwarding to external link

Reply:
done